runtime/scripts/part.js

```diff
@@ -1532,6 +1550,7 @@ if(res) { \
         var result;
         result = this.mark_answer(studentAnswer,scope, exec_path);
         if(result.waiting_for_pre_submit) {
+            this.events.trigger('mark', mark);
```

Comment:
There is no `mark` variable.

This should be triggered at the top of the function, with `scope` and `exec_path` as arguments.